packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -242,21 +293,17 @@ export class PartialSequenceLengths {
                 if (seqLTE(segment.seq, collabWindow.minSeq)) {
                     combinedPartialLengths.minLength += segment.cachedLength;
                 } else {
-                    if (segment.seq !== UnassignedSequenceNumber) {
-                        PartialSequenceLengths.insertSegment(combinedPartialLengths, segment);
-                    }
+                    PartialSequenceLengths.insertSegment(combinedPartialLengths, segment);
                 }
                 const removalInfo = toRemovalInfo(segment);
                 if (seqLTE(removalInfo?.removedSeq, collabWindow.minSeq)) {
                     combinedPartialLengths.minLength -= segment.cachedLength;
-                } else {
-                    if (removalInfo !== undefined
-                        && removalInfo.removedSeq !== UnassignedSequenceNumber) {
-                        PartialSequenceLengths.insertSegment(
-                            combinedPartialLengths,
-                            segment,
-                            removalInfo);
-                    }
+                } else if (removalInfo !== undefined ||
+                    (segment.localRemovedSeq !== undefined && segment.localRemovedSeq !== UnassignedSequenceNumber)) {
```

Comment:
s
```suggestion
                    (segment.localRemovedSeq !== undefined && segment.removedSeq !== UnassignedSequenceNumber)) {
```
localRemovedSeq should never be UnassignedSequenceNumber it will always be an increasing integer

Reply:
the check is also a big redundant, as if there is a localRemovedSeq then the removeSeq should always be UnassignedSequenceNumber, but i think the verbosity is fine for now, as it helps illustrate the intent

Reply:
thanks, good catch.

> the check is also a bit redundant, as if there is a localRemovedSeq then the removeSeq should always be UnassignedSequenceNumber, but i think the verbosity is fine for now, as it helps illustrate the intent

I actually had to change this behavior in #11316 in order to get this PR to work. Overlapping remove with local, un-acked now keeps the segment in a state with localRemovedSeq as well as removedSeq. Some more details on that PR.